setup.py

```diff
@@ -25,8 +33,18 @@
     long_description = f.read()
 
 
-# the extension is optional since in case of lack of c the api
-# there is a ctypes fallback and a slow python fallback
+class BuildExt(base_build_ext):
```

Comment:
Is this code copied from somewhere, the Python 3.x optional=true code or something? Just curious.

Reply:
its a small reimplementation of the python3 mechanism, which has a context managers and a few more bits to sort it out